GVRf/Framework/jni/engine/renderer/renderer.cpp

```diff
@@ -504,66 +500,83 @@ void Renderer::renderRenderData(RenderData* render_data,
+                    glm::mat4 mvp_matrix(projection_matrix * mv_matrix);
+                    try {
+                        bool right = render_mask & RenderData::RenderMaskBit::Right;
+                        switch (curr_material->shader_type()) {
+                        case Material::ShaderType::UNLIT_SHADER:
+                            shader_manager->getUnlitShader()->render(mvp_matrix,
+                                    render_data, curr_material);
+                            break;
+                        case Material::ShaderType::UNLIT_HORIZONTAL_STEREO_SHADER:
+                            shader_manager->getUnlitHorizontalStereoShader()->render(
+                                    mvp_matrix, render_data, curr_material,
+                                    right);
+                            break;
+                        case Material::ShaderType::UNLIT_VERTICAL_STEREO_SHADER:
+                            shader_manager->getUnlitVerticalStereoShader()->render(
+                                    mvp_matrix, render_data, curr_material,
+                                    right);
+                            break;
+                        case Material::ShaderType::OES_SHADER:
+                            shader_manager->getOESShader()->render(mvp_matrix,
+                                    render_data, curr_material);
+                            break;
+                        case Material::ShaderType::OES_HORIZONTAL_STEREO_SHADER:
+                            shader_manager->getOESHorizontalStereoShader()->render(
+                                    mvp_matrix, render_data, curr_material,
+                                    right);
+                            break;
+                        case Material::ShaderType::OES_VERTICAL_STEREO_SHADER:
+                            shader_manager->getOESVerticalStereoShader()->render(
+                                    mvp_matrix, render_data, curr_material,
+                                    right);
+                            break;
+                        case Material::ShaderType::CUBEMAP_SHADER:
+                            shader_manager->getCubemapShader()->render(
+                                    model_matrix, mvp_matrix, render_data,
+                                    curr_material);
+                            break;
+                        case Material::ShaderType::CUBEMAP_REFLECTION_SHADER:
+                            shader_manager->getCubemapReflectionShader()->render(
+                                    mv_matrix, glm::inverseTranspose(mv_matrix),
+                                    glm::inverse(view_matrix), mvp_matrix,
+                                    render_data, curr_material);
+                            break;
+                        default:
+                            shader_manager->getCustomShader(
+                                    curr_material->shader_type())->render(
```

Comment:
Yes you are right. The name gives some confusion for the case of custom shaders now.  
I agree that the signature was not changed with this pull request. But, I am seeing a crash when there is a call, other than the base pass, for example pass 1. The shader type or shader ID was not initialize correctly somehow. Do you see similar to this on your device?:
Fatal signal 6 (SIGABRT) at 0x000033af (code=-6), thread 13268 (OVR::VrThread)
pid: 13231, tid: 13268, name: OVR::VrThread  >>> org.gearvrf.gvroutlinesample <<<